srv/modules/runners/changed.py

```diff
@@ -155,20 +156,27 @@ def help_():
     return ""
 
 
-def requires_conf_change(service, cluster='ceph'):
+def requires_conf_change(**kwargs):
     """
     If any of the dependent services received a change
     in the its config, retrun True
     """
-    cfg = Config(service)
+    cfg = Config(**kwargs)
+    service = cfg.service_name
+    cluster = kwargs.get('cluster', 'ceph')
+    if not service:
+        # or raise?
+        return False
     # pylint: disable=invalid-name
     local = salt.client.LocalClient()
     if service not in cfg.dependencies:
         return "Service {} not defined".format(service)
     for deps in cfg.dependencies[service]:
-        if Config(deps).has_change():
+        if Config(service=deps).has_change():
```

Comment:
That has bug potential. Dependencies are loaded dynamically from a hash. By them time you initialize the `Config` class you don't know where to look for `conf_dir`. Need to maintain the meta information alongside the service name.